src/main/java/org/jetbrains/kotlinx/lincheck/runner/ParallelThreadsRunner.kt

```diff
@@ -280,9 +280,10 @@ open class ParallelThreadsRunner(
     }
 
      // For [TestThreadExecution] instances
-    class TestThread(r: Runnable) : Thread(r) {
+    class TestThread(r: Runnable, runner: ParallelThreadsRunner) : Thread(r) {
         var iThread: Int = 0
         var cont: CancellableContinuation<*>? = null
+        val hashCode = runner.hashCode()
```

Comment:
Let's rename `hashCode` to just `hash` and pass it as a parameter (instead of passing `ParallelThreadsRunner`)